lib/OpenQA/Schema/Result/JobGroups.pm

```diff
@@ -338,28 +338,33 @@ sub to_template {
 
         foreach my $product (sort keys %{$group{scenarios}->{$arch}}) {
             my @scenarios;
-            foreach my $test_suite (@{$group{scenarios}->{$arch}->{$product}}) {
-                foreach my $name (sort keys %$test_suite) {
-                    my $attr = $test_suite->{$name};
-                    if ($attr->{machine} eq $default_machine) {
-                        delete $attr->{machine} if $test_suites{$arch}{$name} == 1;
-                    }
-                    if (keys %$attr) {
-                        $test_suite->{$name} = $attr;
-                        push @scenarios, $test_suite;
-                    }
-                    else {
-                        push @scenarios, $name;
-                    }
-                }
-            }
-            $group{scenarios}{$arch}{$product} = \@scenarios;
+            _remove_test_suite_defaults($product, $default_machine, $arch, \%group, \%test_suites, @scenarios);
         }
     }
 
     return \%group;
 }
 
+sub _remove_test_suite_defaults {
+    my ($product, $default_machine, $arch, %group, %test_suites, @scenarios) = @_;
```

Comment:
That's not going to work. You cannot initialize multiple hashes and arrays from the argument list like that. Also, use signatures.

```suggestion
sub _remove_test_suite_defaults ($product, $default_machine, $arch, $group, $test_suites, $scenarios) {
```

Of course this means you need to adjust the calling site to pass the hashes/arrays not directly but only a reference. The code inside the function also needs to be adjusted accordingly.